v2/internal/reconcilers/arm/azure_generic_arm_reconciler.go

```diff
@@ -70,37 +71,72 @@ func NewAzureDeploymentReconciler(
 		PositiveConditions: positiveConditions,
 		Config:             cfg,
 		Extension:          extension,
+		ARMOwnedResourceReconcilerCommon: reconcilers.ARMOwnedResourceReconcilerCommon{
+			ResourceResolver: resourceResolver,
+			ReconcilerCommon: reconcilers.ReconcilerCommon{
+				KubeClient:         kubeClient,
+				PositiveConditions: positiveConditions,
+			},
+		},
 	}
 }
 
-func (r *AzureDeploymentReconciler) Reconcile(
-	ctx context.Context,
-	log logr.Logger,
-	eventRecorder record.EventRecorder,
-	obj genruntime.MetaObject) (ctrl.Result, error) {
-
+func (r *AzureDeploymentReconciler) asARMObj(obj genruntime.MetaObject) (genruntime.ARMMetaObject, error) {
 	typedObj, ok := obj.(genruntime.ARMMetaObject)
 	if !ok {
-		return ctrl.Result{}, errors.Errorf("cannot modify resource that is not of type ARMMetaObject. Type is %T", obj)
+		return nil, errors.Errorf("cannot modify resource that is not of type ARMMetaObject. Type is %T", obj)
 	}
 
+	return typedObj, nil
+}
+
+func (r *AzureDeploymentReconciler) makeInstance(log logr.Logger, eventRecorder record.EventRecorder, obj genruntime.MetaObject) (*azureDeploymentReconcilerInstance, error) {
+	typedObj, err := r.asARMObj(obj)
+	if err != nil {
+		return nil, err
+	}
 	// Augment Log with ARM specific stuff
 	log = log.WithValues("azureName", typedObj.AzureName())
 
 	// TODO: The line between AzureDeploymentReconciler and azureDeploymentReconcilerInstance is still pretty blurry
```

Comment:
Do we still need this TODO?

Reply:
Yeah I think this TODO is still warranted.